example/m5_competition.py

```diff
@@ -0,0 +1,311 @@
+"""
+The M5 Competition, held in 2020, was part of the prestigious Makridakis
+Forecasting Competitions. The goal of this competition was to accurately
+forecast the sales of 3,000 individual items across 10 Walmart stores for the
+next 28 days. The winning strategy involved transforming the complex
+multi-variate time series dataset into a more manageable tabular format, which
+was then used to train a collection of Gradient Boosted Trees models.
+
+See https://www.kaggle.com/competitions/m5-forecasting-accuracy/overview.
+
+In this example, we will explore how to use the Temporian library to replicate
+this transformation from time series into a tabular dataset. By following the
+step-by-step guide provided, you will learn how to leverage the library's
+powerful functionalities to replicate transformations of your time series data.
+With Temporian, converting time series data into a tabular format has never been
+easier.
+"""
+
+import os
+import temporian as tp
+import pandas as pd
+import numpy as np
+from datetime import datetime, timezone, timedelta
+import urllib.request
+import zipfile
+import tempfile
+
+# Directory used to download the raw M5 dataset and to export the tabular
+# dataset.
+work_directory = os.path.join(tempfile.gettempdir(), "m5")
+os.makedirs(work_directory, exist_ok=True)
+print("Use work directory:", work_directory)
+
+# Download the M5 dataset
+raw_data_zip = os.path.join(work_directory, "raw.zip")
+if not os.path.exists(raw_data_zip):
+    print("Download M5 dataset in", raw_data_zip)
+    url = "https://docs.google.com/uc?export=download&id=1NYHXmgrcXg50zR4CVWPPntHx9vvU5jbM&confirm=t"
+    urllib.request.urlretrieve(url, raw_data_zip)
+
+# Extract the M5 dataset
+raw_data_dir = os.path.join(work_directory, "raw")
+if not os.path.exists(raw_data_dir):
+    print("Extract M5 dataset in", raw_data_dir)
+    with zipfile.ZipFile(raw_data_zip, "r") as zip_ref:
+        zip_ref.extractall(raw_data_dir)
+
+
+# Load raw dataset
+print("Load raw dataset")
+print("================")
+
+# During development, set nrows to a small value (e.g. nrows=100) to
+# only load a sample of the dataset.
+#
+# TODO: Set to None when fast enough.
+nrows = 100
+
+raw_path = lambda x: os.path.join(raw_data_dir, x)

```

Comment:
Is there a reason to avoid `pathlib`? I guess we'd save some lines here and by doing just `savefig(work_directory / "raw_sales.png")` below.

Reply:
Ack